swarm/storage/resource_test.go

```diff
@@ -0,0 +1,245 @@
+package storage
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/binary"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"golang.org/x/net/idna"
+
+	"github.com/ethereum/go-ethereum/contracts/ens"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+var (
+	blockCount = uint64(4200)
+)
+
+func init() {
+	log.Root().SetHandler(log.CallerFileHandler(log.LvlFilterHandler(log.LvlTrace, log.StreamHandler(os.Stderr, log.TerminalFormat(true)))))
+}
+
+type FakeRPC struct {
+	blockcount *uint64
+}
+
+func (r *FakeRPC) BlockNumber() (string, error) {
+	return strconv.FormatUint(*r.blockcount, 10), nil
+}
+
+func TestResourceValidContent(t *testing.T) {
+	// privkey for signing updates
+	privkey, err := crypto.GenerateKey()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// temp datadir
+	datadir, err := ioutil.TempDir("", "rh")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(datadir)
+
+	rh, err := NewResourceHandler(privkey, datadir, &testCloudStore{}, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	validname, err := idna.ToASCII("føø.bar")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	key := rh.resourceHash(ens.EnsNode(validname), 4200, 1)
+	chunk := NewChunk(key, nil)
+
+	data := make([]byte, 8)
+	_, err = rand.Read(data)
+	if err != nil {
+		t.Fatal(err)
+	}
+	rh.hasher.Reset()
+	rh.hasher.Write(data)
+	datahash := rh.hasher.Sum(nil)
+	sig, err := crypto.Sign(datahash, privkey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	chunk.SData = make([]byte, 8+SIGNATURE_LENGTH)
+	copy(chunk.SData[:SIGNATURE_LENGTH], sig)
+	copy(chunk.SData[SIGNATURE_LENGTH:], data)
+
+	// TODO: change this to verifyContent on ENS integration
+	recoveredaddress, err := rh.getContentAccount(chunk.SData)
+	if err != nil {
+		t.Fatal(err)
+	}
+	originaladdress := crypto.PubkeyToAddress(privkey.PublicKey)
+
+	if recoveredaddress != originaladdress {
+		t.Fatalf("addresses dont match: %x != %x", originaladdress, recoveredaddress)
+	}
+}
+
+func TestResourceHandler(t *testing.T) {
+
+	// privkey for signing updates
```

Comment:
Again - extracting the setup code for the tests (getting all the boilerplate in 1-2 methods), would make it easier to reason about what the test is supposed to test. From line 96 to 134.